openedx/stanford/djangoapps/auth_lagunita/social.py

```diff
@@ -0,0 +1,67 @@
+from django.contrib.auth.models import User
+from django.shortcuts import redirect
+from social_core.pipeline import partial
+
+from third_party_auth import provider
+from third_party_auth.pipeline import AUTH_DISPATCH_URLS
+from third_party_auth.pipeline import AUTH_ENTRY_LOGIN
+from third_party_auth.pipeline import AUTH_ENTRY_REGISTER
+
+
+def _dispatch_to_login():
+    """
+    Redirects to the login page
+    """
+    return redirect(AUTH_DISPATCH_URLS[AUTH_ENTRY_LOGIN])
+
+
+def _dispatch_to_register():
+    """
+    Redirects to the registration page
+    """
+    return redirect(AUTH_DISPATCH_URLS[AUTH_ENTRY_REGISTER])
+
+
+@partial.partial
+def try_send_existing_lead_to_login(
+        strategy,
+        auth_entry,
+        backend=None,
+        user=None,
+        social=None,
+        current_partial=None,
+        allow_inactive_user=False,
+        *args,
+        **kwargs
+):
+    def should_force_account_creation():
```

Comment:
Cribbed from existing platform code.

It is embedded inside another function there, so it wasn't practical to import it here,
thus the redefinition.